frontend/javascripts/oxalis/model/volumetracing/volumelayer.js

```diff
@@ -277,79 +208,77 @@ class VolumeLayer {
     const state = Store.getState();
     const { brushSize } = state.userConfiguration;
+    const dimIndices = Dimensions.getIndices(this.plane);
 
-    const radius = Math.round(brushSize / 2);
-    const width = 2 * radius;
-    const height = 2 * radius;
+    const unzoomedRadius = Math.round(brushSize / 2);
+    const width = Math.floor((2 * unzoomedRadius) / this.activeResolution[dimIndices[0]]);
+    const height = Math.floor((2 * unzoomedRadius) / this.activeResolution[dimIndices[1]]);
 
-    const map = new Array(width);
-    for (let x = 0; x < width; x++) {
-      map[x] = new Array(height).fill(false);
-    }
-    const floatingCoord2d = this.get2DCoordinate(position);
-    const coord2d = [Math.floor(floatingCoord2d[0]), Math.floor(floatingCoord2d[1])];
-    const minCoord2d = [coord2d[0] - radius, coord2d[1] - radius];
+    const map = new Uint8Array(width * height).fill(0);
+
+    const floatingCoord2d = this.get2DCoordinate(
+      scaleGlobalPositionWithResolutionFloat(position, this.activeResolution),
+    );
+    const minCoord2d = [
+      Math.floor(floatingCoord2d[0] - width / 2),
+      Math.floor(floatingCoord2d[1] - height / 2),
+    ];
 
     // Use the baseVoxelFactors to scale the circle, otherwise it'll become an ellipse
     const [scaleX, scaleY] = this.get2DCoordinate(
       getBaseVoxelFactors(state.dataset.dataSource.scale),
     );
 
     const setMap = (x, y) => {
-      map[x][y] = true;
+      map[x * height + y] = 1;
     };
-    Drawing.fillCircle(radius, radius, radius, scaleX, scaleY, setMap);
-
-    const numberOfSlices = getNumberOfSlicesForResolution(activeResolution, this.plane);
-    const thirdDimensionIndex = Dimensions.thirdDimensionForPlane(this.plane);
+    Drawing.fillCircle(
+      Math.floor(unzoomedRadius / this.activeResolution[dimIndices[0]]),
+      Math.floor(unzoomedRadius / this.activeResolution[dimIndices[1]]),
+      unzoomedRadius,
+      scaleX / this.activeResolution[dimIndices[0]],
+      scaleY / this.activeResolution[dimIndices[1]],
+      setMap,
+    );
 
-    const iterator = new VoxelIterator(
+    const buffer2D = new VoxelBuffer2D(
       map,
       width,
       height,
       minCoord2d,
       this.get3DCoordinate.bind(this),
-      thirdDimensionIndex,
-      numberOfSlices,
-      boundings,
+      this.getFast3DCoordinateFunction(),
     );
-    return iterator;
+    return buffer2D;
   }
 
   drawOutlineVoxels(setMap: (number, number) => void, mode: VolumeTool): void {
     const contourList = this.getContourList();
     const state = Store.getState();
+    const dimIndices = Dimensions.getIndices(this.plane);
     const [scaleX, scaleY] = this.get2DCoordinate(
       getBaseVoxelFactors(state.dataset.dataSource.scale),
     );
     const { brushSize } = state.userConfiguration;
-    const radius = Math.round(brushSize / 2);
+    const radius = Math.round(brushSize / 2); // todo: wont work for anisotropic mags
```

Comment:
todo comment

Reply:
Ah, thanks for raising this. The comment is outdated, since the scale is adapted a few lines below. I'll remove it!